docs/developer/design/volume_v1.0.rst

```diff
@@ -0,0 +1,294 @@
+Volume management v1.0
+======================
+
+* MetalK8s-Version: 2.4
+* Replaces:
+* Superceded-By:
+
+
+Absract
+-------
+
+To be able to run stateful services (such as Prometheus, Zenko or Hyperdrive),
+MetalK8s needs the ability to provide and manage persistent storage resources.
+
+To do so we introduce the concept of MetalK8s **Volume**, using a **Custom
+Resource Definition** (CRD), built on top of the existing concept of
+**Persistent Volume** from Kubernetes. Those **Custom Resources** (CR) will be
+managed by a dedicated Kubernetes operator which will be responsible for the
+storage preparation (using Salt states) and lifetime management of the backing
+**Persistent Volume**.
+
+Volume management will be available from the Platform UI (through a dedicated
+tab under the node page). There, users will be able to create, monitor and
+delete MetalK8s volumes.
+
+
+Scope
+-----
+
+The scope of this first version of Volume Management will be minimalist but
+still functionally useful.
+
+
+Goal
+^^^^
+
+* support two kinds of **Volume**: **sparseLoopDevice** (backed by a sparse
+  file) and **rawBlockDevice** (using whole disk)
+* add support for volume creation (one by one) in the Platform UI
+* add support for volume deletion (one by one) in the Platform UI
+* add support for volume listing/monitoring (show status, size, …) in the
+  Platform UI
+* document how to create a volume
+* document how to create a storage class
+* automated tests on volume workflow (creation, deletion, …)
+
+
+Non-Goal
+^^^^^^^^
+
+* RAID support
+* LVM support
+* expose raw block device (unformated) as **Volume**
+* use an **Admission Controller** for semantic validation
+* auto-discovery of the disks
+* batch provisioning from the UI
+
+
+Proposal
+--------
+
+To implement this feature we need to:
+
+* define and deploy a new CRD describing a MetalK8s Volume
+* develop and deploy a new Kubernetes operator to manage the MetalK8s volumes
```

Comment:
should we use `Volumes` or `volumes`?

Reply:
I would say it depends if we talk about the concept or the object.

Line above use `Volume` because the CRD describe the schema of the object (but I'll use bold font to make it clearer)